c10/cuda/CUDAException.h

```diff
@@ -70,6 +80,13 @@ class C10_CUDA_API CUDAError : public c10::Error {
 // diagnostic if it didn't.
 #define C10_CUDA_KERNEL_LAUNCH_CHECK() C10_CUDA_CHECK(cudaGetLastError())
 
+/// Number of assertion failure messages we can store. If this is too small
+/// threads will fail silently.
+// We use a preprocessor macro here so the value will also be available in
+// the CUDA code.
+#define C10_CUDA_DSA_ASSERTION_COUNT 10
+#define C10_CUDA_DSA_MAX_STR_LEN 512
```

Comment:
Why those needs to be preprocessor macros rather than constexprs?

Reply:
I don't think it'll work with `constexpr`, but we can try.

Reply:
Looks like it worked. Nice catch.